<?xml version="1.0"?>

<launch>
    <param name="/board_height" type="double" value="0.1157" />
    <node pkg="robothon2023" type="task_sm.py" name="task_sm" output="screen">
        <rosparam command="load" file="$(find robothon2023)/config/boundary_safety.yaml"/>
        <rosparam command="load" file="$(find robothon2023)/config/task_params.yaml"/>
        <rosparam command="load" file="$(find robothon2023)/config/trajectories.yaml"/>
        <rosparam command="load" file="$(find robothon2023)/config/winding_poses_new.yaml"/>
        <rosparam command="load" file="$(find robothon2023)/config/joint_angles.yaml"/>
        <rosparam command="load" file="$(find robothon2023)/config/probe_action.yaml"/>
        <rosparam command="load" file="$(find robothon2023)/config/door_open_trajectories.yaml"/>
        <rosparam command="load" file="$(find robothon2023)/config/byod_poses.yaml"/>
        <remap from="~board_detector_event_out" to="/task_board_detector/event_out" />
        <remap from="~board_detector_event_in" to="/task_board_detector/event_in" />
        <remap from="~poi_event_in" to="/points_of_interest_publisher/event_in" />
        <remap from="~init_approximate_board_pose" to="/task_board_detector/approximate_board_pose" />
    </node>
    <include file="$(find robothon2023)/launch/task_board_detector.launch" />
</launch>

